Build requires non-public dependencies?

As of commit fad2a6d32ce8d5314c6daafa699f80985b865719 (the one helpfully titled "cb6a5c67d8805288f33f8eb5e6b27f3a13ee7d5f" :smiley:) I still can't build snarky. The errors are as follows:

```[ERROR] The compilation of snarky failed at "jbuilder build -p snarky -j 4".                                                                                                                          
                                                                                                                                                                                                      
#=== ERROR while installing snarky.~unknown ===================================#                                                                                                                      
# opam-version 1.2.2                                                                                                                                                                                  
# os           linux                                                                                                                                                                                  
# command      jbuilder build -p snarky -j 4                                                                                                                                                          
# path         /home/snarky/.opam/system/build/snarky.~unknown                                                                                                                                        
# compiler     system (4.05.0)                                                                                                                                                                        
# exit-code    1                                                                                                                                                                                      
# env-file     /home/snarky/.opam/system/build/snarky.~unknown/snarky-19-186c2e.env                                                                                                                   
# stdout-file  /home/snarky/.opam/system/build/snarky.~unknown/snarky-19-186c2e.out                                                                                                                   
# stderr-file  /home/snarky/.opam/system/build/snarky.~unknown/snarky-19-186c2e.err                                                                                                                   
### stderr ###                                                                                                                                                                                        
# [...]                                                                                                                                                                                               
# x - caml_mnt6_specific.cpp.o                                                                                                                                                                        
# x - pedersen.cpp.o                                                                                                                                                                                  
# x - tbcs.cpp.o                                                                                                                                                                                      
# x - memory_store_trace.cpp.o                                                                                                                                                                        
# x - ra_memory.cpp.o                                                                                                                                                                                 
# x - fooram_aux.cpp.o                                                                                                                                                                                
# x - tinyram_aux.cpp.o                                                                                                                                                                               
# ~/.opam/system/build/snarky.~unknown/_build/default/src/camlsnark_c/libsnark-caml/build ~/.opam/                                                                                                    
system/build/snarky.~unknown/_build/default/src/camlsnark_c/libsnark-caml ~/.opam/system/build/sna                                                                                                    
rky.~unknown/_build/default/src/camlsnark_c                                                                                                                                                           
# ~/.opam/system/build/snarky.~unknown/_build/default/src/camlsnark_c/libsnark-caml ~/.opam/system                                                                                                    
/build/snarky.~unknown/_build/default/src/camlsnark_c                                                                                                                                                 
# ~/.opam/system/build/snarky.~unknown/_build/default/src/camlsnark_c  
```

/home/snarky/.opam/system/build/snarky.~unknown/snarky-19-186c2e.err says:
```
File "src/jbuild", line 6, characters 26-39:
Error: Library "bitstring_lib" not found.
Hint: try: dune external-lib-deps --missing -p snarky @install
        bash src/camlsnark_c/libcamlsnark_c_stubs.a
/home/snarky/.opam/system/build/snarky.~unknown/_build/default/src/camlsnark_c
~/.opam/system/build/snarky.~unknown/_build/default/src/camlsnark_c/libsnark-caml ~/.opam/system/build/snarky.~unknown/_build/default/src/camlsnark_c                                                
~/.opam/system/build/snarky.~unknown/_build/default/src/camlsnark_c/libsnark-caml/build ~/.opam/system/build/snarky.~unknown/_build/default/src/camlsnark_c/libsnark-caml ~/.opam/system/build/snarky.~unknown/_build/default/src/camlsnark_c
```

That command reveals a second missing dependency:
```
snarky@d65716b40f52:/source$ dune external-lib-deps --missing -p snarky @install
Error: The following libraries are missing in the default context:
- bitstring_lib
- interval_union
Hint: try: opam install bitstring_lib interval_union
```

And those packages don't seem to exist.
```
snarky@d65716b40f52:/source$ opam install bitstring_lib interval_union
[ERROR] No package named interval_union found.
[ERROR] No package named bitstring_lib found.
```

Admittedly, I'm new to OCaml packaging but this does seem to more-or-less follow your README instructions.

--- a/Dockerfile
+++ b/Dockerfile
@@ -20,6 +20,7 @@ RUN apt-get -q update && \
         ocaml \
         opam \
         nano \
+        pkg-config \
         python-markdown && \
     apt-get clean && \
     rm -rf /var/lib/apt/lists/*
@@ -30,39 +31,29 @@ RUN useradd snarky -m
 # Create a volume we can work in. For initial build, we'll copy the local
 # context. To update the snarky library itself later, bind mount your updated
 # source over this and run the build again.
-RUN mkdir -p /source && chown -R snarky:snarky /source
+ADD . /source
+RUN chown -R snarky:snarky /source
 VOLUME ["/source"]
 
 # Be the new user before initializing OPAM.
 USER snarky
-WORKDIR /home/snarky/
 
 # Move to a newer version of OCaml and install dune/jbuilder.
 RUN opam init -y && \
     opam switch $OCAML_VERSION && \
     opam install dune
 
-# Actually copy the source files here, for caching reasons.
-ADD . /source
-
-USER snarky
+WORKDIR /source
 
 # Pin and install the dependencies.
-RUN cd /source && \
-    eval `opam config env` && \
+RUN eval `opam config env` && \
     opam pin add -y interval_union .
-RUN cd /source && \
-    eval `opam config env` && \
+
+RUN eval `opam config env` && \
     opam pin add -y bitstring_lib .
 
-RUN cd /source && \
-    eval `opam config env` && \
-    opam pin add -y snary .
-
-# Do the build!
 RUN eval `opam config env` && \
-    opam install -y snarky
+    opam pin add -y snarky .
 
 # Use a slight hack to always have the current OCaml environment.
 CMD ["/bin/bash", "--init-file", "/home/snarky/.opam/opam-init/init.sh"]
-

